mmpose/models/data_preprocessors/data_preprocessor.py

```diff
@@ -97,3 +157,92 @@ def _get_pad_shape(self, data: dict) -> List[tuple]:
                             'or a tuple with inputs and data_samples, but got'
                             f'{type(data)}: {data}')
         return batch_pad_shape
+
+    def preprocess(self,
+                   data: dict,
+                   training: bool = False) -> Union[dict, list]:
+
+        data = self.cast_data(data)  # type: ignore
+        _batch_inputs = data['inputs']
+        # Process data with `pseudo_collate`.
+        if is_seq_of(_batch_inputs, torch.Tensor):
+            batch_inputs = []
+            for _batch_input in _batch_inputs:
+                # channel transform
+                if self._channel_conversion:
+                    _batch_input = _batch_input[[2, 1, 0], ...]
+
+                if self.normalize_bakend == 'cv2':
+                    _batch_input = _batch_input.float()
+                elif self.normalize_bakend == 'pillow':
+                    _batch_input_array = _batch_input.detach().cpu().numpy(
+                    ).transpose(1, 2, 0)
+                    assert _batch_input_array.dtype == np.uint8, \
+                        'Pillow backend only support uint8 type'
+                    pil_image = Image.fromarray(_batch_input_array)
+                    _batch_input = tvF.to_tensor(pil_image).to(
+                        _batch_input.device)
+
+                # Normalization.
+                if self._enable_normalize:
+                    if self.mean.shape[0] == 3:
+                        assert _batch_input.dim(
+                        ) == 3 and _batch_input.shape[0] == 3, (
+                            'If the mean has 3 values, the input tensor '
+                            'should in shape of (3, H, W), but got the tensor '
+                            f'with shape {_batch_input.shape}')
+                    if self.normalize_bakend == 'cv2':
+                        _batch_input = (_batch_input - self.mean) / self.std
+                    elif self.normalize_bakend == 'pillow':
+                        _batch_input = tvF.normalize(
+                            _batch_input, mean=self.mean, std=self.std)
```

Comment:
`normalize_backend` seems to be confusing, since the normalization is actually conducted by `torchvision`.
If we want to emphasize the img is loaded by `pillow`, maybe `imdecode_backend` is more explicit, just like mmcv does https://github.com/open-mmlab/mmcv/blob/b361a81ad4c5d31611dad7e8d5fbcb7d7df22f34/mmcv/transforms/loading.py#L33